web/src/message_edit_history.ts

```diff
@@ -18,30 +21,61 @@ import * as timerender from "./timerender";
 import * as ui_report from "./ui_report";
 import {user_settings} from "./user_settings";
 
-export function fetch_and_render_message_history(message) {
-    channel.get({
+type ContentEditHistoryItem = {
+    timestamp: string;
+    display_date: string;
+    show_date_row: boolean;
+    edited_by_notice: string;
+    body_to_render?: string;
+    topic_edited?: boolean;
+    prev_topic?: string;
+    new_topic?: string;
+    stream_changed?: boolean;
+    prev_stream?: string;
+};
```

Comment:
I think you want to call this `EditHistoryEntry`; `ContentEditHistoryEntry` could be a fine name for a subtype that's a content-only edit, but I'm not sure why we'd want `content` in the name otherwise. (And `entry` matches naming used in the server for the same concept, I think).

Reply:
Sure, that sounds good!

Reply:
For what it's worth, "item" is used several times in this file. Would it be better to rename those as well to `prev_stream_entry` and `entry`?

Reply:
Probably would be a fine follow-up, yeah.